fibers/_pyfibers.py

```diff
@@ -50,13 +50,25 @@ def _run(c):
             raise error('parent cannot be on a different thread')
         self.parent = parent
 
+    def _get_active_parent(self):
+        parent = self.parent
+        while True:
+            if (parent is not None
+                    and parent._cont is not None
+                    and not parent._ended):
+                break
```

Comment:
Thanks! Will fix it tomorrow. Can you just mention my username so it shows up in my notifications?